### ISSUE TITLE
Filter transform broken after 0.17.0 update

### ISSUE THREAD

**Issue by birkoffe** [vectordotdev/vector#9666]
After updating from `vector-0.16.1-x86_64-unknown-linux-musl` to `vector-0.17.0-x86_64-unknown-linux-musl` broken filter transform -- not pass valid message.

```toml
[transforms.logstash_filter]
  type = "filter"
  inputs = ["logstash_in"]
  condition = 'exists(.application} && exists(.fname}'
```

Does not find anything in changlelog on there. It's bug?

**Comment by birkoffe** [comment]
Oh, it was #9157... Fixed by changeing from

```toml
[source.lohstash_in]
  type = "http"
  address = "0.0.0.0:8000"

  encoding = "json"
```

to

```toml
[source.lohstash_in]
  type = "http"
  address = "0.0.0.0:8000"

  decoding.codec = "json"
```

**Comment by birkoffe** [comment]
May be need to add this change in upgrade guide?

And old `encoding = "json"` still in Examples https://vector.dev/docs/reference/configuration/sources/http/

**Comment by pablosichert** [comment]
Hey @birkoffe, this is indeed an unintended breaking change that we introduced with `0.17`.

A fix has already been merged with #9666 that we will release shortly in `0.17.1` with #9646.

Thank you for reporting, and sorry for the inconvenience!
